app/assets/v2/js/grants/index.js

```diff
@@ -175,18 +204,23 @@ if (document.getElementById('grants-showcase')) {
         vm.fetchGrants();
 
       },
-      changeQuery: function(query) {
+      changeQuery: debounce(function(query) {
         let vm = this;
 
         vm.fetchedPages = [];
         vm.$set(vm, 'params', {...vm.params, ...query});
 
         if (vm.tabSelected === 'grants') {
+          if (vm.params.keyword && vm.params.sort_option == 'weighted_shuffle') {
+            vm.params.sort_option = '';
+          } else if (vm.params.keyword == '' && vm.params.sort_option != 'weighted_shuffle') {
+            vm.params.sort_option = 'weighted_shuffle';
```

Comment:
This will behave in unexpected ways - when you first add a keyword (providing you have weighted_shuffle selected on the sort) this will correctly clear that sort, but if you were to then add a different selection to the query (say you added a tag filter) we would hit the next if statement which would restore the weighted_shuffle selection.

Consider adding this to the keyword input itself - i.e onkeyup we want to be clearing the sort, any other action should leave the sort selection alone.

Reply:
Ahhh, yes you are correct! My bad, sorry! I still think it would be useful to do the clearing onkeyup, because this will only cover us if the user had weighted_shuffle applied before entering a keyword. We would want to clear the sorting method no matter what the user had previously selected when they enter/alter a keyword.

Reply:
gotcha... just pushed the commit for keyup